src/layers/conv.jl

```diff
@@ -1452,3 +1452,200 @@ function Base.show(io::IO, l::EGNNConv)
+[Attention, Learn to Solve Routing Problems!](https://arxiv.org/abs/1706.03762) paper,
+including a successive feed-forward network as well as skip layers and batch normalization.
+
+The layer's basic forward pass is given by
+```math
+x_i' = W_1x_i + \sum_{j\in N(i)} \alpha_{ij} (W_2 x_j + W_6e_{ij})
+```
+where the attention scores are
+```math
+\alpha_{ij} = \mathrm{softmax}\left(\frac{(W_3x_i)^T(W_4x_j+
+W_6e_{ij})}{\sqrt{d}}\right).
+```
+
+Optionally, a combination of the aggregated value with transformed root node features 
+by a gating mechanism via
+```math
+x'_i = \beta_i W_1 x_i + (1 - \beta_i) \underbrace{\left(\sum_{j \in \mathcal{N}(i)}
+\alpha_{i,j} W_2 x_j \right)}_{=m_i}
+```
+with
+```math
+\beta_i = \textrm{sigmoid}(W_5^{\top} [ W_1 x_i, m_i, W_1 x_i - m_i ]).
+```
+can be performed.
+
+# Arguments 
+
+- `in`: Dimension of input features, which also corresponds to the dimension of 
+    the output features.
+- `ein`: Dimension of the edge features; if 0, no edge features will be used.
+- `heads`: Number of heads in output.
+- `concat`: Concatenate layer output or not. If not, layer output is averaged
+    over the heads.
+- `init`: Weight matrices' initializing function.
+- `add_self_loops`: Add self loops to the input graph.
+- `bias_qkv`: If set, bias is used in the key, query and value transformations for nodes.  
+- `bias_root`: If set, the layer will also learn an additive bias for the root when root 
+    weight is used.
+- `root_weight`: If set, the layer will add the transformed root node features
+    to the output.
+- `beta`: If set, will combine aggregation and transformed root node features by
+    gating mechanism.
+- `skip_connection`: If set, a skip connection will be made from the input and 
+    added to the output.
+- `batch_norm`: If set, a batch normalization will be applied to the output.
+- `ff_channels`: If positive, a feed-forward NN is appended, with the first having the given
+    number of hidden nodes; this NN also gets a skip connection and batch normalization 
+    if the respective parameters are set.
+
+"""
+struct TransformerConv{TW1, TW2, TW3, TW4, TW5, TW6, TFF, TBN1, TBN2} <: GNNLayer
+    W1::TW1
+    W2::TW2
+    W3::TW3
+    W4::TW4
+    W5::TW5
+    W6::TW6
+    FF::TFF
+    BN1::TBN1
+    BN2::TBN2
+    channels::Pair{NTuple{2,Int},Int}
+    heads::Int
+    add_self_loops::Bool
+    concat::Bool
+    skip_connection::Bool
+    sqrt_out::Float32
+end
+
+@functor TransformerConv
+
+Flux.trainable(l::TransformerConv) = (l.W1, l.W2, l.W3, l.W4, l.W5, l.W6, l.FF, l.BN1, l.BN2)
+
+TransformerConv(ch::Pair{Int,Int}, args...; kws...) = TransformerConv((ch[1], 0) => ch[2], args...; kws...)
+
+function TransformerConv(ch::Pair{NTuple{2, Int}, Int}; 
+    heads::Int=1, concat::Bool=true, init=glorot_uniform, add_self_loops::Bool=false, 
+    bias_qkv=true, bias_root::Bool=true, root_weight::Bool=true, beta::Bool=false, 
+    skip_connection::Bool=false, batch_norm::Bool=false, ff_channels::Int=0)
```

Comment:
```suggestion
        heads::Int = 1, 
        concat::Bool = true, 
        init = glorot_uniform, 
        add_self_loops::Bool = false, 
        bias_qkv = true, 
        bias_root::Bool = true, 
        root_weight::Bool = true, 
        beta::Bool = false, 
        skip_connection::Bool = false, 
        batch_norm::Bool = false, 
        ff_channels::Int = 0)
```

Reply:
also maybe `beta=true` would be a better default so that we match the paper

Reply:
Thanks for polishing. Concerning the default for beta, I will argue below.